pkg/services/hcloud/server/server.go

```diff
@@ -189,6 +189,15 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 	return res, nil
 }
 
+func handleRateLimit(hm *infrav1.HCloudMachine, err error, functionName string, errMsg string) error {
+	if hm.Status.Ready != true {
```

Comment:
currently, the logic is wrong. What if the error is different than "rate limit exceeded"?